File: CS/CustomItemsSample/CustomItems/SankeyChart/SankeyItemMetadata.cs
using System.ComponentModel;
using DevExpress.DashboardCommon;

namespace CustomItemsSample {
    [
    DisplayName("Sankey"),
    CustomItemDescription("Sankey description"),
    CustomItemImage("DashboardMainDemo.Images.SankeyCustomItem.svg")
    ]
    public class SankeyItemMetadata : CustomItemMetadata {
        [
        DisplayName("Weight"),
        EmptyDataItemPlaceholder("Weight")
        ]
        public Measure Weight {
            get { return GetPropertyValue<Measure>(); }
            set { SetPropertyValue(value); }
        }
        [
        DisplayName("Target"),
        EmptyDataItemPlaceholder("Target"),
        SupportColoring(DefaultColoringMode.None),
        SupportInteractivity
        ]
        public Dimension Target {
            get { return GetPropertyValue<Dimension>(); }
            set { SetPropertyValue(value); }
        }
        [
        DisplayName("Source"),
        EmptyDataItemPlaceholder("Source"),
        SupportColoring(DefaultColoringMode.Hue),
        SupportInteractivity
        ]
        public Dimension Source {
            get { return GetPropertyValue<Dimension>(); }
            set { SetPropertyValue(value); }
        }
    }
}

File: VB/CustomItemsSample/CustomItems/SankeyChart/SankeyItemMetadata.vb
Imports System.ComponentModel
Imports DevExpress.DashboardCommon

Namespace CustomItemsSample
	<DisplayName("Sankey"), CustomItemDescription("Sankey description"), CustomItemImage("CustomItemsSample.Images.SankeyCustomItem.svg")>
	Public Class SankeyItemMetadata
		Inherits CustomItemMetadata

		<DisplayName("Weight"), EmptyDataItemPlaceholder("Weight")>
		Public Property Weight() As Measure
			Get
				Return GetPropertyValue(Of Measure)()
			End Get
			Set(ByVal value As Measure)
				SetPropertyValue(value)
			End Set
		End Property
		<DisplayName("Target"), EmptyDataItemPlaceholder("Target"), SupportColoring(DefaultColoringMode.None), SupportInteractivity>
		Public Property Target() As Dimension
			Get
				Return GetPropertyValue(Of Dimension)()
			End Get
			Set(ByVal value As Dimension)
				SetPropertyValue(value)
			End Set
		End Property
		<DisplayName("Source"), EmptyDataItemPlaceholder("Source"), SupportColoring(DefaultColoringMode.Hue), SupportInteractivity>
		Public Property Source() As Dimension
			Get
				Return GetPropertyValue(Of Dimension)()
			End Get
			Set(ByVal value As Dimension)
				SetPropertyValue(value)
			End Set
		End Property
	End Class
End Namespace
